sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java

```diff
@@ -379,4 +379,25 @@ private void releaseOnNotSubscribedResponse(ReactorNettyResponseState reactorNet
         CANCELLED,
         ERROR;
     }
+
+    /**
+     * DO NOT USE
+     *
+     * This API is only for testing purposes, don't use it anywhere else in the code.
+     * This changes the logging level of Reactor Netty Http Client.
+     */
+    public void enableNetworkLogging() {
+        Logger logger = LoggerFactory.getLogger(REACTOR_NETWORK_LOG_CATEGORY);
+        if (logger.isTraceEnabled()) {
+            this.httpClient = this.httpClient.wiretap(REACTOR_NETWORK_LOG_CATEGORY, LogLevel.TRACE);
```

Comment:
Can't we do it in the test, we can use reflection to get httpClient  out of ReactorNettyClient and then do the same thing there ?

Reply:
We can, but wanted to do it cleanly in one place. Also, this API will be used for further improvement for Reactor Netty Client logging. Current implementation has some restrictions on what logging customer can use for this Http Client. So added this API which will be used in future for updating the Reactor netty http client logging options.

Reply:
I would introduce it in main code when needed for customer use. Right now we can create a method in some base test class and use every where else

Reply:
In my opinion, using some reflection to update the config options of http client is not clean way. We will need to use reflection to get internal `httpClient` from `ReactorNettyClient` which just complicates this simple test. 

Also, `ReactorNettyClient` is already an implementation class, so creating this API is not going to affect public surface area in any way.

Reply:
No strong opinion , so please go ahead with current. All set

Reply:
Thanks @simplynaveen20